src/ui/common/src/components/integrations/dialogs/dialog.tsx

```diff
@@ -40,147 +49,58 @@ import { RedshiftDialog } from './redshiftDialog';
 import { isS3ConfigComplete, S3Dialog } from './s3Dialog';
 import { SnowflakeDialog } from './snowflakeDialog';
 
-type AddTableDialogProps = {
+type Props = {
   user: UserProfile;
-  integrationId: string;
+  service: Service;
   onCloseDialog: () => void;
-  onConnect: () => void;
+  onSuccess: () => void;
+  integrationToEdit?: Integration;
 };
 
-export const AddTableDialog: React.FC<AddTableDialogProps> = ({
+const IntegrationDialog: React.FC<Props> = ({
   user,
-  integrationId,
+  service,
   onCloseDialog,
-  onConnect,
+  onSuccess,
+  integrationToEdit = undefined,
 }) => {
-  const [config, setConfig] = useState<CSVConfig>({
-    name: '',
-    csv: null,
-  });
-  const [disableConnect, setDisableConnect] = useState(true);
-  const [successMessage, setSuccessMessage] = useState('');
-  const [showSuccessToast, setShowSuccessToast] = useState(false);
-  const [isConnecting, setIsConnecting] = useState(false);
-  const [errMsg, setErrMsg] = useState(null);
-
-  const handleSuccessToastClose = () => {
-    setShowSuccessToast(false);
-  };
-
-  useEffect(() => {
-    setDisableConnect(!isConfigComplete(config));
-  }, [config]);
-
-  const dialogHeader = (
-    <Box
-      sx={{
-        display: 'flex',
-        flexDirection: 'row',
-        justifyContent: 'space-between',
-        width: '100%',
-      }}
-    >
-      <Typography variant="h5">{'Upload CSV'}</Typography>
-      <img
-        height="45px"
-        src={
-          'https://spiral-public-assets-bucket.s3.us-east-2.amazonaws.com/webapp/pages/integrations/csv-outline.png'
-        }
-      />
-    </Box>
+  const editMode = !!integrationToEdit;
+  const dispatch: AppDispatch = useDispatch();
+  const [config, setConfig] = useState<IntegrationConfig>(
+    editMode
+      ? { ...integrationToEdit.config } // make a copy to avoid accessing a state object
```

Comment:
Why exactly are we doing this instead of just integrationToEdit.config?

I can see that integrationToEdit is a prop above, but i'm not sure yet how this is being used as a component.

(This should be fine, just curious for my own edification)

Reply:
Yeah it's a good question. This `integrationToEdit.config` is from the redux store for the current integration. The integration detail page renders this value.

In this way we need a copy of this state for the user to mess up during editing (so that the page won't re-render the value user's trying to update). Otherwise any input will directly get reflected on the current integration